lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -506,6 +508,23 @@ spec = do
                             || not (rangeHasUpperBound slotRange)
                             || upperBoundSucc slotRange == slotRange)
 
+        let f sps r = fmap (slotStartTime sps) <$> slotRangeFromTimeRange sps r
+        let sharedDescription = " (where f = fmap slotStartTime . \
+            \slotRangeFromTimeRange)"
+        it ("f timeRange ⊆ timeRange" ++ sharedDescription) $
+            withMaxSuccess 1000 $ property $
+                \sps timeRange -> let r = getUniformTime <$> timeRange in
+                    cover 10 (isJust $ f sps r) "slot range exists" $ do
+                    case f sps r of
+                        Just r' -> r' `isSubrangeOf` r
+                        Nothing -> True
+        it ("f^n === f" ++ sharedDescription) $
```

Comment:
What do you think about the following name:

```hs
it "(fmap (fmap slotStartTime) . slotRangeFromTimeRange) is idempotent"
```
One advantage is that the user doesn't have to scroll up to determine what `f` is, as the description is completely self-contained.

Another advantage is that "idempotent" already has a well-defined meaning here.

What do you think?

(I've omitted the `SlotParameter` arguments for conciseness.)

Reply:
I've changed this to "`slotStartTime . slotRangeFromTimeRange` is idempotent". We're using `.` in a very hand-wavy way, but testing the composition of these two functions is indeed our intent.